### DISCUSSION TITLE
Reset-Password api  generate a post req for generate link for update password to keycloak  having 403 error

### DISCUSSION THREAD

**Discussion by ParthaPratima** [Sunbird-Ed/Community#477] [Bugs] [answered]
---Reset-Password api  generate a post req for generate link for update password to keyclaok  having 403 error-----------------------------------------------------------------------------------------------------------------------------------------------------------

POST  :  http://localhost:8080/auth/realms/<realm name>/get-required-action-link

{
    "redirectUri": "https://dev.sunbirded.org/resources",
    "clientId": "<client name>",
    "expirationInSecs": "155520000",
    "userName": <user name>",
    "requiredAction": "UPDATE_PASSWORD"
}

ERROR:: 
{
    "errorMessage": "Does not have realm admin role."
}

**Comment by Abhign** [comment]
@reshmi-nair @Hari-stackroute @AmiableAnil Can you please check this error.

CC: @Krishnaj20 @chitranshu-keshav @vinukumar-vs

**Reply by ParthaPratima** [reply]
Thanks @Abhign for the reply,it would be of great help if the team helps me in resolving the issue,it has become a blocker for me currently and have already spent some time on this,I have logged in as admin in the keycloak console and have also checked the roles section to see if the user has admin role and it looks fine.
Below the token information(token used to hit /auth/realms/get-required-action-link) after decoding it.And i see admin role as well
{
  "jti": "0f82f573-5c21-4ccd-bee8-0373d2dc8953",
  "exp": 1682682422,
  "nbf": 0,
  "iat": 1682678822,
  "iss": "http://localhost:8080/auth/realms/sunbird",
  "aud": "realm-management",
  "sub": "55d4c2da-e31f-4d6f-ba56-51ee1aa4a3bd",
  "typ": "Bearer",
  "azp": "sunbird",
  "auth_time": 0,
  "session_state": "ad4bab22-7ea6-4ab1-a69c-109abab6300b",
  "acr": "1",
  "realm_access": {
    "roles": [
      "offline_access"
    ]
  },
  "resource_access": {
    "realm-management": {
      "roles": [
        "view-identity-providers",
        "view-realm",
        "manage-identity-providers",
        "impersonation",
        "realm-admin",
        "create-client",
        "manage-users",
        "query-realms",
        "view-authorization",
        "query-clients",
        "query-users",
        "manage-events",
        "manage-realm",
        "view-events",
        "view-users",
        "view-clients",
        "manage-authorization",
        "manage-clients",
        "query-groups"
      ]
    },
    "sunbird": {
      "roles": [
        "realm-admin",
        "admin"
      ]
    }
  },
  "scope": "profile email",
  "email_verified": true,
  "preferred_username": "sunbirduser2"
}
Hoping for a positive response at the earliest.
Thanks,
Pratha

**Reply by Krishnaj20** [reply]
@rhwarrier - Can someone from Lern BB help here?

@reshmi-nair @Hari-stackroute @AmiableAnil

**Reply by ParthaPratima** [reply]
"Hi  @Krishnaj20 , thank you for your suggestion.  I'm hoping someone from Lern BB might be able to provide some guidance.  Any help would be greatly appreciated!"

**Reply by reshmi-nair** [reply, marked as answer]
@ParthaPratima Please refer to the thread [LernDiscussions #111](https://github.com/orgs/Sunbird-Lern/discussions/111) for reply. Lets keep it open and close this thread and also [https://github.com/orgs/Sunbird-Ed/discussions/471](https://github.com/orgs/Sunbird-Ed/discussions/471) because these are duplicates for same issue.

**Reply by chitranshu-keshav** [reply]
@ParthaPratima Can you please mark this thread as answered
